src/bin/cql-stress-cassandra-stress/settings/option/mode.rs

```diff
@@ -0,0 +1,212 @@
+
+use crate::settings::{
+    param::{
+        types::{ConnectionsPerHost, ConnectionsPerShard},
+        ParamsParser, SimpleParamHandle,
+    },
+    ParsePayload,
+};
+
+pub struct ModeOption {
+    pub compression: Option<Compression>,
+    pub user_credentials: Option<(String, String)>,
+    pub pool_size: PoolSize,
+}
+
+impl ModeOption {
+    pub const CLI_STRING: &'static str = "-mode";
+
+    pub fn description() -> &'static str {
+        "CQL connection options"
+    }
+
+    pub fn parse(cl_args: &mut ParsePayload) -> Result<Self> {
+        let params = cl_args.remove(Self::CLI_STRING).unwrap_or_default();
+        let (parser, handles) = prepare_parser();
+        parser
+            .parse(params)
+            .context("Failed to parse -mode option parameters")?;
+        Self::from_handles(handles)
+    }
+
+    pub fn print_help() {
+        let (parser, _) = prepare_parser();
+        parser.print_help();
+    }
+
+    pub fn print_settings(&self) {
+        println!("Mode:");
+        println!("  Compression: {:?}", self.compression);
+        if let Some((username, password)) = &self.user_credentials {
+            println!("  Username: {}", username);
+            println!("  Password: {}", password);
+        }
+        println!("  Pool size: {:?}", self.pool_size);
+    }
+
+    fn from_handles(handles: ModeParamHandles) -> Result<ModeOption> {
+        let compression = handles.compression.get().unwrap();
+        let user_credentials = match (handles.username.get(), handles.password.get()) {
+            (Some(username), Some(password)) => Some((username, password)),
+            (None, None) => None,
+            _ => {
+                return Err(anyhow::anyhow!(
+                    "Both username and password need to be either set or unspecified"
+                ))
+            }
+        };
+        let pool_size = match handles.connections_per_shard.get() {
+            Some(per_shard) => per_shard,
+            None => handles.connections_per_host.get().unwrap(),
+        };
+
+        Ok(Self {
+            compression,
+            user_credentials,
+            pool_size,
+        })
+    }
+}
+
+struct ModeParamHandles {
+    compression: SimpleParamHandle<Option<Compression>>,
+    username: SimpleParamHandle<String>,
+    password: SimpleParamHandle<String>,
+    connections_per_host: SimpleParamHandle<ConnectionsPerHost>,
+    connections_per_shard: SimpleParamHandle<ConnectionsPerShard>,
+}
+
+fn prepare_parser() -> (ParamsParser, ModeParamHandles) {
+    let mut parser = ParamsParser::new(ModeOption::CLI_STRING);
+
+    // `cql3` and `native` parameters are ignored but we introduce them so we support
+    // the cassandra-stress CLI syntax which is used in SCT.
+    let _cql3 = parser.simple_param::<bool>("cql3", None, "cql3 mode (dummy parameter)", false);
+    let _native =
+        parser.simple_param::<bool>("native", None, "native mode (dummy parameter)", false);
```

Comment:
Why do those variable names start with underscores? Usually it is done for variables that are not used, but they are used in this function.

Reply:
While I get it that the _parameters_ are not used in the program, the variables themselves are used.

Reply:
I wanted to emphasize the fact that these parameters are ignored and not used by the runtime. But I see your point and I understand the intent behind the names starting with underscore - I'll remove the underscores then.